pkg/util/span/frontier.go

```diff
@@ -109,7 +109,7 @@ func (r OpResult) asBool() bool {
 type Operation func(roachpb.Span, hlc.Timestamp) (done OpResult)
 
 var useBtreeFrontier = envutil.EnvOrDefaultBool("COCKROACH_BTREE_SPAN_FRONTIER_ENABLED",
-	metamorphic.ConstantWithTestBool("COCKROACH_BTREE_SPAN_FRONTIER_ENABLED", false))
+	metamorphic.ConstantWithTestBool("COCKROACH_BTREE_SPAN_FRONTIER_ENABLED", true))
```

Comment:
hold up: changing this implies that on 24.3, we'll use the btree frontier right? Should we keep it on for roachtests for a few months before releasing this?

Reply:
We had it behind a metamorphic for a year now, and had it on by default for a few months already, before I flipped it off in the spring when I tracked down this most recent bug, so anything the roachtests were gonna find they already found. I don't think we learn anything from a delay.

Reply:
good point. Seems fine to merge it now. Hopefully all these scale tests + drt bake time will catch anything else.